### PR TITLE
[CORE] Support JDK 11

### PR DESCRIPTION
**Pull request by surnaik**
## What changes were proposed in this pull request?

* Add jdk-11 maven profile
* Add jdk11 tpcds/tpch github action workflow
* Update docs
* Upgrades Caffeine cache version for Java 11 and 17

## How was this patch tested?

Pass CI

### CODE REVIEW

**Comment by surnaik** [comment]
@ulysses-you Requesting your review since you upgraded to Java 17. Thanks!

**Comment by surnaik** [comment]
Please take a look - @zhouyuan @yaooqinn Thanks!!

**Review by zhouyuan — COMMENTED** [review]
Looks good to me, is it ok to limit the tests to Spark 3.5 first, so we could save some CI resource?

**Comment by ulysses-you** [comment]
+1 for @zhouyuan to reduce the CI resource

File: .github/workflows/velox_docker.yml [hunk 1]
```diff
@@ -510,6 +533,8 @@ jobs:
       - name: Setup java and maven
         run: |
           apt-get update && apt-get install -y openjdk-8-jdk maven wget
+          apt remove openjdk-11* -y
+          export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
```
Comment (zhztheplayer) [inline .github/workflows/velox_docker.yml:537]: GHA creates new shell for each step so this export command may not work. You can try using `>> $GITHUB_ENV` to store the variables.

Comment (surnaik) [reply .github/workflows/velox_docker.yml]: Thanks, fixed!!

**Comment by surnaik** [comment]
@zhouyuan @ulysses-you I have now enabled it only for Spark 3.5, Please take a look at this again. Thanks!

**Pull request merged**